common/utils.h

```diff
@@ -20,6 +20,10 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#ifdef __cplusplus
```

Comment:
I still don't see a change here to utils.h having stuff outside the header guard?